tests/js/server/aql/aql-persistent-index.js

```diff
@@ -30,20 +30,64 @@ const internal = require("internal");
 const jsunity = require("jsunity");
 const helper = require("@arangodb/aql-helper");
 const getQueryResults = helper.getQueryResults;
-  
+const isCluster = require('internal').isCluster();
+
 const cn = "UnitTestsCollection";
-  
+const idxName = "testIdx";
+const query1 = `FOR d IN ${cn} FILTER d.value == 1 SORT d._key RETURN d`;
+const query2 = `FOR d IN ${cn} FILTER d.value == 2 SORT d._key RETURN d`;
+
+
 let explain = function (query, params) {
   return helper.removeClusterNodes(helper.getCompactPlan(AQL_EXPLAIN(query, params, { optimizer: { rules: [ "-all", "+use-indexes" ] } })).map(function(node) { return node.type; }));
 };
 
+let testandValidate = function (query, isUnique, insertMoreDocs, insertIndexBeforeDocs, sameValue, isPrimary, selectivity) {
+  let ensureIndex = () => {
+    internal.db[cn].ensureIndex({type: "persistent", name: idxName, unique: isUnique, fields: ["value"]});
+  };
+
+  if (insertMoreDocs) {
+    if (insertIndexBeforeDocs) {
+      ensureIndex();
+    }
+    let docs = [];
+    for (let i = 0; i < 10000; ++i) {
+      docs.push({value: (sameValue ? 2 : i + 10)});
+      if (docs.length >= 1000) {
+        internal.db[cn].insert(docs);
+        docs = [];
+      }
+    }
+    if (!insertIndexBeforeDocs) {
+      ensureIndex();
+    }
+  } else {
+    ensureIndex();
+  }
+  require("internal").waitForEstimatorSync();
+  const nodes = AQL_EXPLAIN(query).plan.nodes;
+  for (const key in nodes) {
+    if (nodes[key].type === "IndexNode") {
+      const indexNode = nodes[key];
+      assertEqual(indexNode.indexes.length, 1);
+      const usedIndex = indexNode.indexes[0];
+      assertEqual(usedIndex.type, (isPrimary ? "primary" : "persistent"));
+      assertEqual(usedIndex.name, (isPrimary ? "primary" : idxName));
+      assertEqual(usedIndex.fields.length, 1);
+      assertEqual(usedIndex.fields[0], (isPrimary ? "_key" : "value"));
+      assertEqual(usedIndex.selectivityEstimate, selectivity);
```

Comment:
If there is no IndexNode in the plan, none of the checks will be executed and the test will still pass.